ga4gh/datamodel/reads.py

```diff
@@ -91,6 +91,124 @@ def setFlag(flagAttr, flag):
         return flagAttr | flag
 
 
+class AlignmentDataMixin(datamodel.PysamDatamodelMixin):
+    """
+    Mixin class that provides methods for getting read alignments
+    from bam files
+    """
+    def _getReadAlignments(
+            self, reference, start=None, end=None, readGroup=None):
+        """
+        Returns an iterator over the specified reads
+        """
+        # TODO If reference is None, return against all references,
+        # including unmapped reads.
+        samFile = self.getFileHandle(self._samFilePath)
+        referenceName = reference.getLocalId().encode()
+        # TODO deal with errors from htslib
+        start, end = self.sanitizeAlignmentFileFetch(start, end)
+        readAlignments = samFile.fetch(referenceName, start, end)
+        for readAlignment in readAlignments:
+            tags = dict(readAlignment.tags)
+            if readGroup is None:
+                if 'RG' in tags:
+                    alignmentReadGroupId = tags['RG']
+                    alignmentReadGroup = self._readGroupLocalIdMap[
+                        alignmentReadGroupId]
+                    yield self.convertReadAlignment(
+                        readAlignment, alignmentReadGroup)
+                else:
+                    # TODO what to do for unmapped reads (i.e. no
+                    # alignment read group context)? skip for now
+                    pass
```

Comment:
What are we using the ReadGroup object for in this case during translation? It mignt be helpful to break up the conversion process into a number of steps that can be composed. It seems that the majority of the translation process only needs the read object. Can we break it up based on that?